src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const getObjectCopy = { ...state };
+  const result = [];
+
+  for (const actionObject of actions) {
+    switch (actionObject.type) {
+      case 'addProperties':
+        for (const key in actionObject.extraData) {
+          getObjectCopy[key] = actionObject.extraData[key];
+        }
+        result.push({ ...getObjectCopy });
```

Comment:
You perform this push in all branches.
Don't repeat yourself - do it once after `switch/case` block